normalizer.go

```diff
@@ -0,0 +1,179 @@
+package sqllexer
+
+import (
+	"regexp"
+	"strings"
+)
+
+type SQLNormalizerConfig struct {
+	DBMS string `json:"dbms"`
+
+	// CollectTables specifies whether the normalizer should also extract the table names that a query addresses
+	CollectTables bool `json:"collect_tables"`
+
+	// CollectCommands specifies whether the normalizer should extract and return commands as SQL metadata
+	CollectCommands bool `json:"collect_commands"`
+
+	// CollectComments specifies whether the normalizer should extract and return comments as SQL metadata
+	CollectComments bool `json:"collect_comments"`
+
+	// KeepSQLAlias reports whether SQL aliases ("AS") should be truncated.
+	KeepSQLAlias bool `json:"keep_sql_alias"`
+}
+
+type NormalizedInfo struct {
+	Tables   []string
+	Comments []string
+	Commands []string
+}
+
+type SQLNormalizer struct {
+	config *SQLNormalizerConfig
+}
+
+func NewSQLNormalizer(config *SQLNormalizerConfig) *SQLNormalizer {
+	return &SQLNormalizer{config: config}
+}
+
+const (
+	ArrayPlaceholder   = "( ? )"
+	BracketPlaceholder = "[ ? ]"
+)
+
+// Normalize takes an input SQL string and returns a normalized SQL string, a NormalizedInfo struct, and an error.
+// The normalizer collapses input SQL into compact format, groups obfuscated values into single placeholder,
+// and collects metadata such as table names, comments, and commands.
+func (n *SQLNormalizer) Normalize(input string) (string, *NormalizedInfo, error) {
```

Comment:
I think the return values should be named. 
```suggestion
func (n *SQLNormalizer) Normalize(input string) (normalized string, info *NormalizedInfo, err error) {
```